feat(playbooks): migration of centurion playbooks to github (#17)
NoFussComputing/ansible_collection_centurion: docs/projects/ansible/collection/centurion/playbooks/teams.md

---
title: Teams

> description: Creation and patching of teams and permissions.

Reviewer: product name is required. this is what search engines uses as part of the page description.

Author: fixed

date: 2024-08-18
template: project.html
about: https://github.com/nofusscomputing/ansible_collection_centurion
---

The teams playbook has been created for the purpose of creating Centurion ERP teams. It allows teams to be defined as configuration as code which allows standardisation of teams and permissions within an organisation. With this playbook it is possible to create every team within an organisation and define the permissions and notes that are to be applied to that team.

The teams playbook includes the [AWX Feature](../../../playbooks/awx.md) to import the playbook as a job template in AWX / Ansible Automation Platform.

The following job template will be created:

*  **Centurion/Access/Teams** Creation and patching of teams and permissions

!!! info 

>     The playbook is able to work with Centurion ERP directly or using the inventory pluggin that is included in this collection.

Reviewer: ```suggestion
    The playbook is able to work with the inventory plugin that is included in this collection.
```

Author: Fixed

## Play workflow

The teams playbook gathers information regarding centurion organisations from the ansible inventory. Using this information the play is designed to create new teams, patch permissions and patch notes. The workflow for the playbook is as follows

- Fetch all organisations from Centurion ERP
- Fetch all existing teams within each organisation from Centurion ERP
- Fetch any teams to be created from inventory
- Create new teams
- Patch all teams with required permissions
- Patch all teams with required notes

## Configuration

> The teams playbook uses variables that are gathered from inventory
> The expected structure of the inventory file is:

Reviewer: ```suggestion
The teams playbook uses variables that are gathered from inventory. The expected structure of the inventory file is:
```

Author: Fixed

```yaml

> centurion_erp:
>   teams:
>     - name: "organisation name"
>       teams:
>       - name: "team-name"
>         permissions: []
>         notes: "permissions must be a list"

Reviewer: ```suggestion

centurion_erp:
  teams:
    - name: "organisation name"
      teams:
        - name: "team-name"
          permissions: []
          notes: "permissions must be a list"
```

Author: Applied change

```

!!! tip "common teams"
    Common teams can be created by using yaml anchors. This is useful when multiple organisations require a common team and permissions to be set.

    ```yaml
    centurion_erp:
      common_teams:
        team_name: &team-name "team_name"
        team_permissions: &team-name-permissions []

>         team_name_notes: &team-name-notes "team_notes"

Reviewer: add a full example that includes specifying the anchors

Author: A full example has been provided now

    ```

!!! info "Upcoming feature"

>     With the release of Centurion ERP V1.1 an API request to fetch all permissions that can be applied will be available. 

Reviewer: this can be omitted as v1.1 will be released within hours of 1.0

Author: This has been removed

---
title: Teams

> description: No Fuss Computings Companion Ansible Collection Centurion teams Playbook.

Reviewer: the what now??

Author: Still need to adjust this

Author: adjusted in commit [cec6cd9](https://github.com/nofusscomputing/ansible_collection_centurion/pull/17/commits/cec6cd91a382d424c008dd89b841499e6299691a)

date: 2024-08-18
template: project.html

> about: https://gitlab.com/nofusscomputing/projects/ansible/collections/kubernetes

Reviewer: fix link to be that of github

Author: fixed

---

> The teams playbook has been created for the purpose of creating standardised teams with a defined set of permissions. With this playbook it is possible to create every team within an organisation and define set teams for newly created team. The playbook can also be used to patch a team with updated permissions or notes.

Reviewer: elaborate or at least add that it enables the teams to be defined as part of config as code

Author: Elaborated

The teams playbook includes the [AWX Feature](../../../playbooks/awx.md) to import the playbook as a job template in AWX / Ansible Automation Platform.

The following job template will be created:

*  **Centurion/Access/Teams** Creation and patching of teams and permissions

On import to AWX / Ansible Automation Platform a credential type will also be created, `Playbook/teams/centurion` that can be used to supply the required secrets and Centurion host.

> !!! info Centurion ERP API

Reviewer: remove the title, it's not required. it'll default to `info`

Author: Done

>     The playbook is designed to use the Centurion ERP API to fetch, create and patch teams and permissions.

Reviewer: the playbook doesn't fetch from centurion. The inventory (centurion) plugin as part of this collection obtains what is required from centurion.

Author: Adjusted

## Play workflow

> The teams playbook gathers information regarding centurion organisations from the ansible inventory and from Centurion itself. using this information the play is designed to create new teams, patch permissions and patch notes. The workflow for the playbook is as follows

Reviewer: > from the ansible inventory and from Centurion itself

the two are one in the same. the inventory plugin within this collection builds the inventory for ansibles consumption.

Author: Think i've applied appropriate text

> - Fetch all organisations

Reviewer: ```suggestion
- Fetch all organisations from centurion
```

Author: fixed

> - Fetch all existing teams within each organisation

Reviewer: ```suggestion
- Fetch all existing teams within each organisation from centurion
```

Author: fixed

- Fetch any teams to be created from inventory

> - Create any new teams that do not exist in centurion

Reviewer: ```suggestion
- Create any new teams that do not exist in centurion
```

the what now????

Author: fixed

- Fetch all teams from all organisations in Centurion
- Patch all teams with required permissions
- Patch all teams with required notes

## Configuration

The teams playbook uses variables that are gathered from inventory
The expected structure of the inventory file is:

```yaml
centurion_erp:
  teams:

    - name: "organisation name"
      teams:
      - name: "team-name"
        permissions: []

>         notes: "permissions must be a list"

Reviewer: tidy this up. spacing or no spacing, choose the one easier to read

Author: Done, went with no spacing

```

> !!! tip common teams

Reviewer: ```suggestion
!!! tip "common teams"
```

titles must be in double quotes

Author: fixed

---
title: Teams

> description: Creation and patching of Centurion ERP teams and permissions 

Reviewer: ```suggestion
description: Ansible Playbook for Creating and patching of Centurion ERP teams, including permissions.
```

Author: fixed in commit [054c201](https://github.com/nofusscomputing/ansible_collection_centurion/pull/17/commits/054c201d1abd6accf7ee94843a472849075bd5e1)

date: 2024-08-18
template: project.html
about: https://github.com/nofusscomputing/ansible_collection_centurion
---

The teams playbook has been created for the purpose of creating Centurion ERP teams. It allows teams to be defined as configuration as code which allows standardisation of teams and permissions within an organisation. With this playbook it is possible to create every team within an organisation and define the permissions and notes that are to be applied to that team.

The teams playbook includes the [AWX Feature](../../../playbooks/awx.md) to import the playbook as a job template in AWX / Ansible Automation Platform.

The following job template will be created:

*  **Centurion/Access/Teams** Creation and patching of teams and permissions

!!! info 

>     The playbook is able to work with the inventory plugin that is included in this collection.

Reviewer: ```suggestion
    The playbook is able to work with the [inventory plugin](../plugins/index.md) that is included in this collection.
```
